src/EventListener/ReadListener.php

```diff
@@ -35,21 +34,19 @@
  */
 final class ReadListener
 {
-    use OperationDataProviderTrait;
-    use ToggleableOperationAttributeTrait;
-
     public const OPERATION_ATTRIBUTE_KEY = 'read';
 
     private $serializerContextBuilder;
+    private $readStage;
 
-    public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider, SubresourceDataProviderInterface $subresourceDataProvider = null, SerializerContextBuilderInterface $serializerContextBuilder = null, IdentifierConverterInterface $identifierConverter = null, ResourceMetadataFactoryInterface $resourceMetadataFactory = null)
+    public function __construct(CollectionDataProviderInterface $collectionDataProvider, ItemDataProviderInterface $itemDataProvider, SubresourceDataProviderInterface $subresourceDataProvider = null, SerializerContextBuilderInterface $serializerContextBuilder = null, IdentifierConverterInterface $identifierConverter = null, ResourceMetadataFactoryInterface $resourceMetadataFactory = null, ReadStageInterface $readStage = null)
```

Comment:
Shouldn't the listener depend only of this new service? And this new service be the first parameter, and passing all other services been deprecated?

Reply:
You're right.